src/platform/linux/x11/window.rs

```diff
@@ -144,45 +136,11 @@ impl XWindow {
 
 impl Drop for XWindow {
     fn drop(&mut self) {
-        // Clear out the window proxy data arc, so that any window proxy objects
-        // are no longer able to send messages to this window.
-        *self.window_proxy_data.lock().unwrap() = None;
-
         // Make sure we return the display to the original resolution if we've changed it
         self.switch_from_fullscreen_mode();
     }
 }
 
-#[derive(Clone)]
-pub struct WindowProxy {
-    data: Arc<Mutex<Option<WindowProxyData>>>,
-}
-
-impl WindowProxy {
-    pub fn wakeup_event_loop(&self) {
```

Comment:
It's really suspicious that this is a dead code.

Reply:
It definitely is right now but maybe it shouldn't be, no idea.

Reply:
reading some code it seems EventsLoopProxy is implemented with different code and this is just a left over from when there was a WindowProxy in the API.